src/sql/migrations.rs

```diff
@@ -1121,6 +1121,18 @@ CREATE INDEX msgs_status_updates_index2 ON msgs_status_updates (uid);
         .await?;
     }
 
+    inc_and_check(&mut migration_version, 127)?;
+    if dbversion < migration_version {
+        // Emails must be moved to `target` or deleted only when `smtp.id` reaches
+        // `target_min_smtp_id` to keep SecureJoin working for multi-device and in case of a local
+        // state loss.
+        sql.execute_migration(
+            "ALTER TABLE imap ADD COLUMN target_min_smtp_id INTEGER NOT NULL DEFAULT 0",
```

Comment:
This also ensures that `v*-request` is fully processed before IMAP deletion. I think we can't rely on that messages are deleted from IMAP without delay, so anyway we have to deal with duplicated `vg-member-added` messages. It was suggested that we could e.g. generate some stable `Message-Id` for them. But at least awaiting for a SecureJoin message to be sent before deleting the processed one guarantees that we never lose state because some device goes offline for long or a single device setup restores from a backup.

Reply:
Ok, you convinced me, so the PR only deletes `vg-request-with-auth` now. Anyway `vg-member-added` may not be delivered as you noted, so rather Bob should do retries on his side as he is a part which wants to join (afair currently we even don't show SecureJoin errors to Alice). Alice losing state isn't really a problem if Bob doesn't lose state. Bob deletes `vg-auth-required`, but that's also not a problem, he can start the protocol again with `vg-request` if he restores from a backup when he had already scanned the Alice's QR code.